src/com/itacademy/eshop/shop/Eshop.java

```diff
@@ -25,8 +26,14 @@ public void printProducts() {
         }
     }
 
-    public void addProduct(Product shirt) {
+    public void addProduct(Product shirt) throws DuplicateProductException {
+        for(Product product : products) {
+            if (shirt.getName() == product.getName()) {
+                throw new DuplicateProductException();
```

Comment:
you can pass message to exception. It would make sense to write it like this:
 throw new DuplicateProductException("product with name: " + product.getName() + "already exists") ;